### PR TITLE
Update to CCCL 2.2.0.

### PR DESCRIPTION
**Pull request by bdice**
This PR updates RAFT to CCCL 2.2.0. Do not merge until all of RAPIDS is ready to update.

Depends on #2048.

Replaces #1464.

### CODE REVIEW

File: ci/build_cpp.sh [hunk 1]
```diff
@@ -9,10 +9,14 @@ export CMAKE_GENERATOR=Ninja
 
 rapids-print-env
 
+LIBRMM_CHANNEL=$(rapids-get-pr-conda-artifact rmm 1404 cpp)
```
Comment (bdice) [inline ci/build_cpp.sh:12]: I will revert changes to CI scripts that use upstream artifacts before merging.

File: fetch_rapids.cmake [hunk 2]
```diff
@@ -11,6 +11,8 @@
 # or implied. See the License for the specific language governing permissions and limitations under
 # the License.
 # =============================================================================
+set(rapids-cmake-repo bdice/rapids-cmake)
+set(rapids-cmake-branch cuco-cccl-2.2.0)
```
Comment (bdice) [inline fetch_rapids.cmake:15]: I will revert this before merging. Depends on https://github.com/NVIDIA/cuCollections/pull/404/ merging, and rapids-cmake being updated to use that commit in `versions.json`.

File: cpp/CMakeLists.txt [hunk 3]
```diff
@@ -177,8 +177,8 @@ include(cmake/modules/ConfigureCUDA.cmake)
 rapids_cpm_init()
 
 if(NOT BUILD_CPU_ONLY)
-  # thrust before rmm/cuco so we get the right version of thrust/cub
-  include(cmake/thirdparty/get_thrust.cmake)
+  # CCCL before rmm/cuco so we get the right version of CCCL
+  include(cmake/thirdparty/get_cccl.cmake)
```
Comment (vyasr) [inline cpp/CMakeLists.txt:181]: Is there a reason that raft can't use the CCCL in rmm? Can you check this? I'm guessing that the end result is equivalent right now, except that by calling this first we end up bundling CCCL into raft _in addition_ to rmm.

**Review by vyasr — CHANGES_REQUESTED** [review]

File: cpp/CMakeLists.txt (hunk 3 above)
Comment (vyasr) [reply cpp/CMakeLists.txt:181]: This diagnosis is correct, but we discussed offline and decided to live with this to allow easier testing of CCCL. The long-term fix will be to switch to a centralized package for CCCL dependencies that will remove the bundling concern in conda.

**Review by vyasr — APPROVED** [review]

**Review by robertmaynard — APPROVED** [review]

File: cpp/CMakeLists.txt [hunk 4]
```diff
@@ -206,7 +206,7 @@ target_include_directories(
 
 if(NOT BUILD_CPU_ONLY)
   # Keep RAFT as lightweight as possible. Only CUDA libs and rmm should be used in global target.
-  target_link_libraries(raft INTERFACE rmm::rmm cuco::cuco nvidia::cutlass::cutlass raft::Thrust)
+  target_link_libraries(raft INTERFACE rmm::rmm cuco::cuco nvidia::cutlass::cutlass CCCL::Thrust)
```
Comment (bdice) [inline cpp/CMakeLists.txt:209]:
```suggestion
  target_link_libraries(raft INTERFACE rmm::rmm cuco::cuco nvidia::cutlass::cutlass CCCL::CCCL)
```

**Pull request merged**